openedx/core/djangoapps/course_groups/models.py

```diff
@@ -49,3 +54,21 @@ class CourseUserGroupPartitionGroup(models.Model):
     )
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
+
+
+class CourseUserGroupSettings(models.Model):
+    """
+    This model represents cohort settings for courses.
+    """
+    is_cohorted = models.BooleanField(default=False)
+
+    course_id = CourseKeyField(
+        unique=True,
+        max_length=255,
+        db_index=True,
+        help_text="Which course is these settings associated with?",
+    )
+
+    cohorted_discussions = models.TextField(null=True, blank=True)  # JSON list
+
+    cohort_inline_discussions = models.BooleanField(default=True)
```

Comment:
Actually we have used `always_cohort_inline_discussions` but due to pylint violation we renamed it. If you want `always_cohort_inline_discussions` than we can disable pylint warning.